doc/thing/thing.rst

```diff
@@ -9,3 +9,4 @@ KNoT Thing
    thing-create-app
    thing-samples
    thing-api
+   thing-supported-boards
```

Comment:
Put before thing-api

Reply:
I think it is more suitable after thing-api since this section contains hardware specifications.